Sonar issues

rdeniel opened the pull request.

rdeniel committed:
SUGGEST-197 Sonar optimizations

rdeniel committed:
SUGGEST-197 Sonar bugs fixes

rdeniel committed:
SUGGEST-197 all the EntryType classes need to get static fields

rdeniel committed:
SUGGEST-197 all variables names not matching conventions fixed

rdeniel committed:
SUGGEST-197 regroup exceptions types

rdeniel committed:
SUGGEST-197 code pratices improvements

rdeniel committed:
SUGGEST-197 code pratices improvements

rdeniel committed:
SUGGEST-197 missing private constructor

rdeniel committed:
SUGGEST-197 remove useless comments

rdeniel committed:
SUGGEST-197 variables assignements were public

rdeniel committed:
SUGGEST-197 duplicate i18n entries

rzara commented on src/java/fr/paris/lutece/plugins/suggest/business/Category.java:
```diff
@@ -100,12 +100,8 @@ public void setTitle( String title )
-        {
-            return true;
-        }
+        return obj != null && obj == this.getClass() && ( (Category) obj ).getIdCategory( ) == _nIdCategory;
```
this seems wrong. `obj` is almost certainly not the class `Category`

rdeniel replied:
Ok, I'll rollback to "obj instanceof Category" instead of "this.getClass()".

rzara commented on src/java/fr/paris/lutece/plugins/suggest/business/SuggestSubmitType.java:
```diff
@@ -185,12 +185,7 @@ public int getIdXSLStyleSheet( )
-        }
-
-        return false;
+        return obj != null && obj == this.getClass() && ( (SuggestSubmitType) obj ).getIdType( ) == _nIdType;
```
this seems wrong. `obj` is almost certainly not the class `SuggestSubmitType`

rdeniel replied:
Ok, I'll rollback to instance of...

rzara commented on src/java/fr/paris/lutece/plugins/suggest/business/VoteTypeHome.java:
```diff
@@ -49,7 +49,7 @@
 
     /**
-     * Private constructor - this class need not be instantiated
+     * Private constructor - this class needs not to be instantiated
```
wouldn't "does not need" be better ?

rdeniel replied:
I suppose so.

rzara commented on src/java/fr/paris/lutece/plugins/suggest/service/search/SuggestIndexer.java:
```diff
@@ -269,11 +259,7 @@ public void indexDocuments( ) throws IOException, InterruptedException, SiteMess
-        catch( TikaException e )
+        catch( SAXException | TikaException e )
         {
             throw new AppException( "Error during page parsing." );
```
Maybe as a drive by correction you should set the AppException Cause here to not loose the details of the error

rdeniel replied:
Ok, I will add the Exception "e" in the AppException's argument.

rzara commented on src/java/fr/paris/lutece/plugins/suggest/utils/ProcessBBCode.java:
```diff
@@ -44,22 +44,6 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-/*
```
Are you sure your are allowed to relicense this code ? This is not a code improvement, and should really by (if the change is appropriate) a separate commit with explicit comments about the license change

rdeniel replied:
Actually, no. I'll restore this.

rzara commented on src/java/fr/paris/lutece/plugins/suggest/utils/SuggestUtils.java:
```diff
@@ -222,7 +222,7 @@ public static void sendNotificationNewSuggestSubmit( Suggest suggest, SuggestSub
         catch( Exception e )
         {
-            AppLogService.error( "Error during Notify new suggest submit  : " + e.getMessage( ) );
+            AppLogService.error( "Error during Notify new suggest submit  : ", e.getMessage( ) );
```
This seems incorrect because you are calling the parametric form of AppLogService.error, but there are no parameter placeholder in the message. You should really pass the exception itself, so that a stack trace is logged and the exception can be diagnosed in the logs

rdeniel replied:
Ok, I'll directly pass the Exception as argument.

rzara commented on src/java/fr/paris/lutece/plugins/suggest/web/SuggestApp.java:
```diff
@@ -1186,29 +1186,12 @@ private void doVote( String strVote, int nIdSubmitSuggest, Plugin plugin, String
         int nScore;
 
-        if ( ( strVote != null ) && strVote.equals( "-2" ) )
+        if ( ( strVote != null ) && strVote.matches( ( "-2|-1|1|2" ) ) )
```
This will compile the regex on each invocation. Maybe the compiled regex should be a static final class member ?

rdeniel replied:
Understood.

rzara commented on src/java/fr/paris/lutece/plugins/suggest/business/SuggestSubmitTypeDAO.java:
```diff
@@ -290,14 +290,9 @@ public boolean isAssociateToSuggest( int nIdType, Plugin plugin )
-    
-                    return true;
-                }
+                return true;
```
Shouldn't you go ahead and just return `daoUtil.next( ) && daoUtil.getInt( 1 ) != 0` like other changes you did ?

rdeniel replied:
Yes, indeed.

rzara commented on src/java/fr/paris/lutece/plugins/suggest/business/attribute/SuggestAttributeHome.java:
```diff
@@ -52,6 +52,14 @@
     private static ISuggestAttributeDAO _dao = SpringContextService.getBean( BEAN_SUGGEST_ATTRIBUTE_DAO );
 
+    /**
+     * Private constructor - this class needs not to be instantiated
```
Wouldn't "does not need to..." be better ?

rdeniel replied:
Yes, certainly.

rdeniel committed:
SUGGEST-197 some best practices corrected

SUGGEST-197 some best practices corrected

SUGGEST-197 some best practices corrected

SUGGEST-197 some best practices corrected

SUGGEST-197 some best practices corrected

SUGGEST-197 some best practices corrected

SUGGEST-197 some best practices corrected

SUGGEST-197 some best practices corrected

rzara commented on src/java/fr/paris/lutece/plugins/suggest/web/SuggestApp.java:
```diff
@@ -1186,7 +1187,7 @@ private void doVote( String strVote, int nIdSubmitSuggest, Plugin plugin, String
         int nScore;
 
-        if ( ( strVote != null ) && strVote.matches( ( "-2|-1|1|2" ) ) )
+        if ( ( strVote != null ) && strVote.matches( DO_VOTE_CASES ) )
```
This does not change the fact that the regexpr is compiled on each invocation. See https://github.com/lutece-platform/lutece-chat-plugin-easyrulesbot/blob/ab7f68990a4568c9bcca9737175a67f5cee0dc5a/src/java/fr/paris/lutece/plugins/easyrulesbot/service/response/processors/InputStringResponseProcessor.java#L50 for an example

rdeniel replied:
Thank you.

rdeniel committed:
SUGGEST-197 some best practices corrected

SUGGEST-197 some best practices corrected

SUGGEST-197 some best practices corrected

rdeniel committed:
SUGGEST-197 some best practices corrected

rdeniel committed:
SUGGEST-197 remove duplicated i18n keys

rdeniel committed:
SUGGEST-197 some best practices corrected

rdeniel committed:
SUGGEST-198 update AppLogService.error calls parameters
